### DISCUSSION TITLE
Bidirectionally syncing annotations with Zotero

### DISCUSSION THREAD

**Discussion by WantToLearnJapanese** [ahrm/sioyek#753] [Q&A] [answered]
Hello!
I am writing Python scripts to sync bidirectionally from Zotero to Sioyek and vice versa.

My question is, is there any hook I can use to capture when Sioyek finishes writing annotations to shared.db? so that I can sync as soon as possible? Or we may able to sync on Sioyek closes a PDF file. 
btw another concern is what if it messes up SQLite DB in some cases like accessing the DB from a syncer and Sioyek simultaneously.

**Comment by ahrm** [comment, marked as answer]
> My question is, is there any hook I can use to capture when Sioyek finishes writing annotations to shared.db?

Not currently.

> btw another concern is what if it messes up SQLite DB in some cases like accessing the DB from a syncer and Sioyek simultaneously.

I think sqlite should handle synchronization and not allow multiple simultaneous writers. Either way, it is entirely possible to mess up the database if you are not careful, so I recommend backing up the database frequently.

**Reply by ryanwwest** [reply]
This syncing is a cool idea.
